tests/v2/validation/rbac/rbac_psa_test.go

```diff
@@ -0,0 +1,336 @@
+package rbac
+
+import (
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/namespaces"
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+	"github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	coreV1 "k8s.io/api/core/v1"
+)
+
+const (
+	containerImage = "nginx"
+	containerName = "psanginx"
+)
+
+type PSATestSuite struct {
+	suite.Suite
+	client              *rancher.Client
+	nonAdminUser        *management.User
+	nonAdminUserClient  *rancher.Client
+	session             *session.Session
+	cluster             *management.Cluster
+	adminProject        *management.Project
+	steveAdminClient    *v1.Client
+	steveNonAdminClient *v1.Client
+	adminNamespace      *v1.SteveAPIObject
+}
+
+func (rb *PSATestSuite) TearDownSuite() {
+	rb.session.Cleanup()
+}
+
+func (rb *PSATestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	rb.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(rb.T(), err)
+
+	rb.client = client
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(rb.T(), clusterName, "Cluster name to install should be set")
+	clusterID, err := clusters.GetClusterIDByName(rb.client, clusterName)
+	require.NoError(rb.T(), err, "Error getting cluster ID")
+	rb.cluster, err = rb.client.Management.Cluster.ByID(clusterID)
+	require.NoError(rb.T(), err)
+
+}
+
+func (rb *PSATestSuite) ConvertNS(namespace *v1.SteveAPIObject) *coreV1.Namespace {
+	getNSSteveObject, err := rb.steveAdminClient.SteveType(namespaces.NamespaceSteveType).ByID(namespace.ID)
+	require.NoError(rb.T(), err)
+
+	namespaceObj := &coreV1.Namespace{}
+	err = v1.ConvertToK8sType(getNSSteveObject.JSONResp, namespaceObj)
+	require.NoError(rb.T(), err)
+
+	return namespaceObj
+}
+
+func (rb *PSATestSuite) ValidatePSA(role string) {
+	labels := map[string]string{
+		psaWarn:    pssPrivilegedPolicy,
+		psaEnforce: pssPrivilegedPolicy,
+		psaAudit:   pssPrivilegedPolicy,
+	}
+
+	rb.T().Logf("Validate updating the PSA labels as %v", role)
+
+	//Testcase1 Validate updating the PSA labels
```

Comment:
Is this comment needed? Seems that we can remove this since you have a log message right above it anyway.